### PR TITLE
[wgsl] Use `alias` instead of `type` for type aliases

### PR DESCRIPTION
**Pull request by FL33TW00D**
This brings us in line with the spec: https://gpuweb.github.io/gpuweb/wgsl/#type-aliases

Currently Canary fails with `type` keyword.

### CODE REVIEW

**Review by ErichDonGubler — APPROVED** [review]
I don't have merge rights, but this LGTM. Let's see who else can approve this. 🙂

File: src/front/wgsl/parse/mod.rs
Comment (jimblandy) [inline src/front/wgsl/parse/mod.rs:2216]: I think it would be best to simply remove support for `type` altogether. Everyone will need to make that change soon eventually.

File: tests/in/boids.wgsl
Comment (jimblandy) [inline tests/in/boids.wgsl:2]: Again, much better to do this in `module-scope.wgsl` or `type-alias.wgsl`, which actually use the definition.

**Review by jimblandy — CHANGES_REQUESTED** [review]
Thanks for fixing this! Just a few minor things.

File: src/front/wgsl/parse/mod.rs
Comment (FL33TW00D) [reply src/front/wgsl/parse/mod.rs]: Agreed!

**Review by teoxoy — APPROVED** [review]
LGTM

**Pull request merged**
